susemanager-utils/susemanager-sls/src/modules/uyuni_config.py

```diff
@@ -0,0 +1,1247 @@
+        log.debug("Create group: %s", name)
+        return self.client("systemgroup.create", name, description)
+
+    def delete(self, name: str) -> int:
+        """
+        Delete a system group.
+
+        :param name: Name of the system group.
+        :return: 1 on success, exception thrown otherwise.
+        """
+        log.debug("delete group: %s", name)
+        return self.client("systemgroup.delete", name)
+
+    def update(self, name: str, description: str) -> Dict[str, Union[int, str]]:
+        """
+        Update an existing system group.
+
+        :param name: Name of the system group.
+        :param description: Description of the system group.
+        :return: data of the system group.
+        """
+        log.debug("update group: %s", name)
+        return self.client("systemgroup.update", name, description)
+
+    def list_systems(self, name: str, minimal: bool = True) -> List[Dict[str, Any]]:
+        """
+        Get information from the system in the group.
+
+        :param name: Group name
+        :param minimal: default True. Minimal information or more detailed one about systems
+        :return: List of system information
+        """
+        return self._convert_datetime_list(
+            self.client("systemgroup.listSystemsMinimal" if minimal else "systemgroup.listSystems", name))
+
+    def add_remove_systems(self, name: str, add_remove: bool, system_ids: List[int] = []) -> int:
+        """
+        Add or remove list of systems from the group
+
+        :param name: Group name
+        :param add_remove: True to add to the group, False to remove
+        :param system_ids: List of system ids to add or remove
+        :return: 1 on success, exception thrown otherwise
+        """
+        return self.client("systemgroup.addOrRemoveSystems", name, system_ids, add_remove)
+
+
+class UyuniSystems(UyuniRemoteObject):
+
+    def get_minion_id_map(self, refresh: bool = False) -> Dict[str, int]:
+        """
+        Map between minion ID and system internal ID of all system user have access to.
+        Context cache can be used, to avoid multiple call to the server.
+
+        :param refresh: Get new data from server, ignoring values in local context cache
+        :return: Map between minion ID and system ID of all system accessible by authenticated user
+        """
+        minions_token_key = "uyuni.minions_id_map_" + self.client.get_user()
+        if (not minions_token_key in __context__) or refresh:
+            __context__[minions_token_key] = self.client("system.getMinionIdMap")
+        return __context__[minions_token_key]
+
+
+class UyuniChildMasterIntegration:
+    """
+    Integration with the Salt Master which is running
+    on the same host as this current Minion.
+    """
+    DEFAULT_MASTER_CONFIG_PATH = "/etc/salt/master"
+
+    class FCkMinions(CkMinions):
+        """
+        Minion data matcher.
+        """
+
+        def _get_key_fingerprint(self, minion_id: str) -> str:
```

Comment:
This method only seems to be called by `_get_fingerprints`, which seems to be dead code.